web/static/js/phoenix_html.js

```diff
@@ -1,13 +1,20 @@
-function didHandleSubmitLinkClick(element){
+function isLinkToSubmitParent(element) {
+  var isLinkTag = element.tagName === 'A';
+  var shouldSubmitParent = element.getAttribute('data-submit') === 'parent';
+
+  return isLinkTag && shouldSubmitParent;
+}
+
+function didHandleSubmitLinkClick(element) {
   while(element) {
-    if(element.matches && element.matches('a[data-submit=parent]')){
+    if(isLinkToSubmitParent(element)) {
       var message = element.getAttribute('data-confirm');
       if (message === null || confirm(message)) {
-        element.parentNode.submit();
+        element.parentElement.submit();
```

Comment:
We need something to prevent this from climbing to the document node, since document isn't an `Element` and doesn't implement `getAttribute`.  Without something to prevent that, it errors any time you click something that isn't an `a[data-confirm=parent]`. Another option would be to set the while loop to `while (element && element.getAttribute)`.